tracer/src/Datadog.Trace/Configuration/ImmutableDynamicSettings.cs

```diff
@@ -0,0 +1,109 @@
+// <copyright file="ImmutableDynamicSettings.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Configuration
+{
+    internal class ImmutableDynamicSettings : IEquatable<ImmutableDynamicSettings>
```

Comment:
Any reason why you didn't use a `record`? wouldn't it give you 90% of this for free? 🤔